linkedin.py

```diff
@@ -161,74 +161,141 @@ def processJob(self, jobID: str, jobCounter: models.JobCounter):
+                jobTitle = self.getJobTitleFromJobCard(jobItem)
+                if not jobTitle:
+                    if config.displayWarnings:
+                        prYellow("⚠️  Couldn't extract job title from job card")
+                    continue
 
-            # TODO Fix when some jobItems don't contain companyNameSpan which is not an expected behavior
-            companyNameSpan = jobItem.find_elements(By.XPATH, ".//span[contains(@class, 'job-card-container__primary-description')]")
-            if companyNameSpan:
-                full_description = companyNameSpan[0].text.strip()
-                
-                textWithinParentheses = utils.extractTextWithinParentheses(full_description)
-                workPlaceType = self.verifyWorkPlaceType(textWithinParentheses)
+                if self.isTitleBlacklisted(jobTitle):
+                    if config.displayWarnings:
+                        prYellow(f"⚠️  Not adding a job as the title '{jobTitle}' is blacklisted")
+                    continue
 
-                if '·' in full_description:
-                    companyName = full_description.split('·')[0].strip()
-                else:
-                    companyName = full_description
+                companyName = self.getCompanyNameFromJobCard(jobItem)
+                if not companyName:
+                    if config.displayWarnings:
+                        prYellow("⚠️  Couldn't extract company name from job card")
+                    continue
 
                 if self.isCompanyBlacklisted(companyName):
                     if config.displayWarnings:
                         prYellow(f"⚠️  Not adding a job as the company '{companyName}' is blacklisted")
                     continue
 
-            if companyName is None:
-                utils.logDebugMessage("Couldn't find companyName", utils.MessageTypes.WARNING)
-                continue
-
-            jobTitleAnchor = jobItem.find_elements(By.XPATH, ".//a[contains(@class, 'job-card-container__link job-card-list__title')]")
-            if len(jobTitleAnchor) > 0:
-                allTexts = jobTitleAnchor[0].text.split("\n")
-                uniqueTexts = list(dict.fromkeys(allTexts))
-                jobTitle = uniqueTexts[0].strip()
-
-                if self.isTitleBlacklisted(jobTitle):
+                jobId = jobItem.get_attribute("data-occludable-job-id")
+                if not jobId:
                     if config.displayWarnings:
-                        prYellow(f"⚠️  Not adding a job as the title '{jobTitle}' is blacklisted")
+                        prYellow("⚠️  Couldn't extract job ID from job card")
                     continue
-            
-            if jobTitle is None:
-                utils.logDebugMessage("Couldn't find jobTitle", utils.MessageTypes.WARNING)
-                continue
-            
-            if jobTitle is None:
-                utils.logDebugMessage("Couldn't find jobTitle", utils.MessageTypes.WARNING)
```

Comment:
This might have been the issue that it was a repeated if statement and something else should have been covered